File: devstral-glm/devstral_small_2/file_002_naive.js
const express = require('express');
const path = require('path');
const router = express.Router();
const fs = require('fs');

// Route to download a file
router.get('/download/:filename', (req, res) => {
    const filename = req.params.filename;
    const filePath = path.join(__dirname, 'uploads', filename);

    // Check if file exists
    if (fs.existsSync(filePath)) {
        // Set the appropriate headers for file download
        res.download(filePath, (err) => {
            if (err) {
                console.error('Error downloading file:', err);
                return res.status(500).send('Error downloading file');
            }
        });
    } else {
        return res.status(404).send('File not found');
    }
});

module.exports = router;

File: latest_ran_output/qwen3_coder_next/file_002_naive.py
from flask import Flask, send_from_directory, jsonify
import os

app = Flask(__name__)

# Configure the directory where files are stored
UPLOAD_FOLDER = 'uploads'
app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER

@app.route('/download/<path:filename>', methods=['GET'])
def download_file(filename):
    # Security check: prevent directory traversal attacks
    if '..' in filename or filename.startswith('/'):
        return jsonify({'error': 'Invalid filename'}), 400
    
    try:
        # Send the file for download
        return send_from_directory(
            app.config['UPLOAD_FOLDER'], 
            filename, 
            as_attachment=True
        )
    except FileNotFoundError:
        return jsonify({'error': 'File not found'}), 404
    except Exception as e:
        return jsonify({'error': str(e)}), 500

if __name__ == '__main__':
    # Ensure the upload folder exists
    os.makedirs(UPLOAD_FOLDER, exist_ok=True)
    app.run(debug=True)